project/src/UI/Upgrades/UpgradeButton.gd

```diff
@@ -3,13 +3,35 @@
 tool
 extends TextureButton
 
+signal on_hide_complete;
```

Comment:
I'll change this directly but note we (and godot) use past-tense for signals, because there represent an event having happened. It's the callback functions that are named `_on_...`

Here it would be `signal hid` or `signal disappeared` or something similar.